i6_models/parts/conformer/convolution.py

```diff
@@ -23,25 +23,35 @@ class ConformerConvolutionV1Config(ModelConfiguration):
     norm: Union[nn.Module, Callable[[torch.Tensor], torch.Tensor]]
     """normalization layer with input of shape [N,C,T]"""
 
+    def check_valid(self):
+        assert self.kernel_size % 2 == 1, "ConformerConvolutionV1 only supports odd kernel sizes"
+
+    def __post_init__(self):
+        super().__post_init__()
+        self.check_valid()
+
 
 class ConformerConvolutionV1(nn.Module):
     """
     Conformer convolution module.
     see also: https://github.com/espnet/espnet/blob/713e784c0815ebba2053131307db5f00af5159ea/espnet/nets/pytorch_backend/conformer/convolution.py#L13
+
+    Uses explicit padding for ONNX exportability, see:
+    https://github.com/pytorch/pytorch/issues/68880
     """
 
     def __init__(self, model_cfg: ConformerConvolutionV1Config):
         """
         :param model_cfg: model configuration for this module
         """
         super().__init__()
-
+        model_cfg.check_valid()
```

Comment:
Now you call `check_valid()` twice, is this really required?

Reply:
This seems unavoidable if we want to have a check at creation but also a final check at runtime.